src/modules/RootPart.rb

```diff
@@ -2138,57 +2054,32 @@ def FindRootPartitions
       @rootPartitions = {}
       @numberOfValidRootPartitions = 0
 
-      # all partitions on all devices
-      max_steps = 0
-      Builtins.foreach(target_map) do |device, description|
-        max_steps = Ops.add(
-          max_steps,
-          Builtins.size(Ops.get_list(description, "partitions", []))
-        )
-      end
+      # all formatted partitions on all devices
+      partitions = probed.filesystems.with { |fs| fs.to_s != "swap" }.partitions
 
       counter = 0
-      Builtins.foreach(target_map) do |device, description|
-        Builtins.foreach(Ops.get_list(description, "partitions", [])) do |partition|
-          counter = Ops.add(counter, 1)
-          if UI.WidgetExists(Id("search_progress"))
-            UI.ChangeWidget(
-              Id("search_pb"),
-              :Value,
-              Ops.divide(Ops.multiply(100, counter), max_steps)
-            )
-          end
-          # some partitions don't make sense at all
-          if Ops.get_symbol(partition, "detected_fs", :unknown) != :swap &&
-              Ops.get_symbol(partition, "type", :primary) != :extended
-            freshman = {}
-
-            if Mode.test
-              freshman = {
-                :valid => true,
-                :name  => "SuSE Linux 4.2",
-                :arch  => "i286",
-                :label => "Label"
-              }
-            else
-              log.debug("Checking partition: #{partition}")
-              freshman = CheckPartition(partition)
-            end
+      partitions.each_with_index do |partition, counter|
+        if UI.WidgetExists(Id("search_progress"))
+          percent = 100 * (counter + 1 / partitions.size)
+          UI.ChangeWidget(Id("search_pb"), :Value, percent)
+        end
 
-            @rootPartitions = Builtins.add(
-              @rootPartitions,
-              Ops.get_string(partition, "device", "error"),
-              freshman
-            )
+        freshman = {}
 
-            if Ops.get_boolean(freshman, :valid, false)
-              @numberOfValidRootPartitions = Ops.add(
-                @numberOfValidRootPartitions,
-                1
-              )
-            end
-          end
+        if Mode.test
```

Comment:
NP: Please drop the `Mode.test` support completely when you are already touching this, the old testsuite has been removed anyway...